text/0042-separate-context-propagation.md

```diff
@@ -96,21 +103,34 @@ To access the context associated with program execution, the Context API provide
 
 ## Propagation API
 
-Distributed applications send data to downstream processes via propagators, functions which read and write application context into RPC requests. Each distributed application creates a set of propagators for every type of supported medium - currently HTTP and Binary.
+Distributed applications propagate their state by data to downstream processes via injectors, functions which read and write application context into RPC requests. Each distributed application creates a set of propagators for every type of supported medium - currently only HTTP requests.
 
-**Inject(context, request)**  
-To send the data for all distributed applications downstream to the next process, the Propagation API provides a function which takes a context and a request, and mutates the request to include the encoded context. The canonical representation of a request is as a map.
+**HTTPInject(context, request)**  
+To send the data for all distributed applications downstream to the next process, the Propagation API provides a function which takes a context and an HTTP request, and mutates the HTTP request to include an HTTP Header representation of the context.
 
-**Extract(context, request) -> context**  
-To receive data injected by prior upstream processes, the Propagation API provides a function which takes a context and a request, and returns an updated context.
+**HTTPExtract(context, request) -> context**  
+To receive data injected by prior upstream processes, the Propagation API provides a function which takes a context and an HTTP request, and returns context which represents the state of the upstream system.
 
-**RegisterPropagator(type, inject, extract)**  
+**ChainHTTPInjector(injector, injector) injector** 
 In order for the application layer to function correctly, Propagation choices must be syncronized between all processes in the distributed system, and multiple applications must be able to inject and extract their context into the same request. To meet these requirements, the Propagation API provides a function which registers a set of propagators, which will all be executed in order when the future calls to inject and extract are made. A canonical propagator consists of an inject and an extract function.
 
-OpenTelemetry currently contains two types of Propagators:
+**ChainHTTPExtractor(extractor, extractor) extractor** 
+In order for the application layer to function correctly, Propagation choices must be syncronized between all processes in the distributed system, and multiple applications must be able to inject and extract their context into the same request. To meet these requirements, the Propagation API provides a function which registers a set of propagators, which will all be executed in order when the future calls to inject and extract are made. A canonical propagator consists of an inject and an extract function.
```

Comment:
Description is identical to ChainHTTPInjector. Perhaps combine them?

Reply:
Hmmm, copypasta error, one moment please...

Reply:
Fixed